### DISCUSSION TITLE
HttpFoundation: The filename fallback must only contain ASCII characters.

### DISCUSSION THREAD

**Discussion by ro0NL** [symfony/symfony#45474] [Q&A]
Hi,

We just got hit by https://github.com/symfony/symfony/blob/60ce5a3dfbd90fad60cd39fcb3d7bf7888a48659/src/Symfony/Component/HttpFoundation/HeaderUtils.php#L167-L173 in prod.

In userland i now use `ResponseHeaderBag::DISPOSITION_ATTACHMENT, $filename, u($filename)->ascii()` arguments everywhere.

But i'm wondering what could be done in core ...

a) remove the feature
b) think of a fallback implementation (without symfony/string)

**Comment by chyshyya** [comment]
What encoding of file contains code line `ResponseHeaderBag::DISPOSITION_ATTACHMENT, $filename, u($filename)->ascii()`?
And show $filename initialization with string as code line.
Which OS used in prod?
